stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -319,6 +319,7 @@ public final class _DataStorage {
     
     @inline(__always)
     public func append(_ bytes: UnsafeRawPointer, length: Int) {
+        precondition(length >= 0, "Length of appending bytes must be positive")
```

Comment:
in terms of integers this should not matter, if we counted bytes with floating point fractional values and could have negative zero then it would make a difference.